code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/filter/serviceevent/ServiceEventFilter.java

```diff
@@ -114,10 +121,23 @@ private boolean isNetworkUp(long accountId) {
         List<Service> services = objectManager.find(Service.class, SERVICE.ACCOUNT_ID, accountId, SERVICE.REMOVED, null, SERVICE.STACK_ID,
                 networkDriverService.getStackId());
         for (Service service : services) {
-            if (!service.getState().equals(CommonStatesConstants.ACTIVE)) {
+            if (!(service.getState().equals(CommonStatesConstants.ACTIVE) && service.getHealthState().equals(HealthcheckConstants.HEALTH_STATE_HEALTHY))) {
                 return false;
             }
         }
         return true;
     }
+
+    private boolean isNetworkStack(long accountId, long instanceId) {
+        Instance instance = objectManager.findAny(Instance.class, INSTANCE.ID, instanceId);
+        List<? extends Service> services = instanceDao.findServicesFor(instance);
+        if(services.size() > 0) {
+            List<Service> network_services = objectManager.find(Service.class, SERVICE.ACCOUNT_ID, accountId, SERVICE.REMOVED, null, SERVICE.STACK_ID,
```

Comment:
You already have the list of services for the instance on line 133, no need to look for them again. What you need to do is - locate the services having kind == network for the instance. You can introduce a new method for that in instanceDao that would accept instanceid and servicekind